src/activities/components/ActivityList.vue

```diff
@@ -426,6 +430,40 @@ export default {
       this.slots = 'all'
       this.type = 'all'
     },
+    checkArchivedTypes () {
+      const archivedTypes = {}
+
+      for (const activityType in this.activityListTypeFilter) {
+        // count the number of archived types and store in a hashmap, defaulting to zero
+        if (this.activityListTypeFilter[activityType].status === 'archived') {
+          archivedTypes[this.activityListTypeFilter[activityType].id] = 0
+        }
+      }
+
+      console.log(this.activities)
+
+      for (const num in this.activities) {
+        // count the number of activities for each archived type and update the hashmap
+        const activity = this.activities[num]
+        if (activity.activityType.id in archivedTypes) {
+          const id = activity.activityType.id
+          archivedTypes[id] += 1
+        }
+      }
+
+      for (const archivedTypeId in archivedTypes) {
+        if (archivedTypes[archivedTypeId] === 0) {
+          // For any archived types with zero activities in view, remove the type from the filter
+          for (const [index, activityType] of this.activityListTypeFilter.entries()) {
+            if (parseInt(activityType.id) === parseInt(archivedTypeId)) {
```

Comment:
Are these not already `int`s?

Reply:
I believe `archivedTypeId` is a string by default (this is just the ID of the archived type), so I convert both values to integers to do a strict comparison and as a safeguard.

Do you think we should convert it to an integer when populating the dictionary, or is this fine the way it is?

Reply:
Ah yeah, it's because you're storing things in a map, and the keys are always strings. So you can't put them in as int's. It would require a different reworking, which I can't explain right now, so it's fine as it is for this purpose :)